ui/runs/forms/data_analysis.py

```diff
@@ -1090,4 +1089,93 @@ def fill_form(self, run: Run) -> None:
             SelectPeptidesForProtein, "peptide_df"
         )
         if single_protein_peptides:
-            self.fields["peptide_df"].initial = single_protein_peptides[0]
\ No newline at end of file
+            self.fields["peptide_df"].initial = single_protein_peptides[0]
+class PowerAnalysisPowerCalculationForm(MethodForm):
+    t_test_results = CustomChoiceField(
+        choices=[],
+        label="T-test results",
+    )
+    #fill alpha dynamic from t-test
+    alpha = CustomFloatField(
+        label="Error rate (alpha)",
+        min_value = 0,
+        max_value = 1,
+        step_size = 0.05,
+        initial = 0.05,
+    )
+    def fill_form(self, run: Run) -> None:
+        self.fields["t_test_results"].choices = get_t_test_results(run)
+
+class PowerAnalysisSampleSizeCalculationForm(MethodForm):
```

Comment:
In general, you should run a code formatter. Often there too few blank lines (between methods or here above the class, you should leave two empty lines) and unnecessary whitespaces around equal signs (as below in the `...Field()`